### PR TITLE
Enable vaapi, to support hardware acceleration

### PR DESCRIPTION
**Pull request by qmfrederik**
This PR adds the `--enable-vaapi` flag, and builds FFmpeg with support for (Intel) hardware acceleration.
The Ubuntu, CentOS and Alpine images have also been updated to include the build-time vaapi dependencies.

### CODE REVIEW

File: templates/Dockerfile-template.ubuntu [hunk 1]
```diff
@@ -61,5 +62,10 @@ ENV         LD_LIBRARY_PATH=/usr/local/lib
 
 COPY --from=build /usr/local /usr/local/
 
+RUN \
+	apt-get update -y && \
+	apt-get install -y libva-drm1 libva1 && \
```
Comment (jrottenberg) [inline templates/Dockerfile-template.ubuntu]: runtime dependency ? 

Just for ubuntu ? That would explain the failure on https://travis-ci.org/jrottenberg/ffmpeg/builds/389354692 though.

**Comment by jrottenberg** [comment]
Thanks a lot for your contribution it was on my todo list for a while and was requested like on #27

**Comment by qmfrederik** [comment]
@jrottenberg I've fixed the CentOS and Alpine builds by installing libva in the target container. Travis shows a couple of errors; that seems to be because of file download issues.

**Comment by jrottenberg** [comment]
It looks great, thank you.

One more commit andI'll merge : https://github.com/jrottenberg/ffmpeg/pull/106/files#diff-c479c96df0fcf8f4474e645d4d473888

File: templates/Dockerfile-template.ubuntu (hunk 1 above)
Comment (GnaphronG) [reply templates/Dockerfile-template.ubuntu]: This should be move into the base container to avoid an extra layer

**Comment by jrottenberg** [comment]
oh !
```
The following NEW packages will be installed:
  i965-va-driver libbsd0 libdrm-amdgpu1 libdrm-common libdrm-intel1
  libdrm-nouveau2 libdrm-radeon1 libdrm2 libedit2 libelf1 libffi6 libllvm5.0
  libpciaccess0 libsensors4 libva-drm1 libva1 libx11-xcb1 libxau6
  libxcb-dri2-0 libxcb-dri3-0 libxcb-present0 libxcb-sync1 libxcb-xfixes0
  libxcb1 libxdmcp6 libxshmfence1 mesa-va-drivers va-driver-all
0 upgraded, 28 newly installed, 0 to remove and 0 not upgraded.
Need to get 15.9 MB of archives.
After this operation, 74.5 MB of additional disk space will be used.
```
I wonder if we should make a variant for that build ...

**Comment by qmfrederik** [comment]
@jrottenberg Using --no-install-recommends seems to fix this:

```
The following additional packages will be installed:
  libdrm-common libdrm2
Recommended packages:
  va-driver-all | va-driver
The following NEW packages will be installed:
  libdrm-common libdrm2 libva-drm1 libva1
0 upgraded, 4 newly installed, 0 to remove and 3 not upgraded.
Need to get 89.0 kB of archives.
After this operation, 356 kB of additional disk space will be used.
```

The recommended part contains the `i695-va-driver`, which needs to run on the host, not inside the Docker container, so I think this is a safe change to make.

**Comment by jrottenberg** [comment]
ok I really want to merge it , but it touches too many images, would you be ok to create a new variant based on ubuntu named :

- jrottenberg/ffmpeg:3.3-vaapi 
- jrottenberg/ffmpeg:4.0-vaapi 
etc

**Pull request merged**